v2/tools/generator/internal/codegen/pipeline/apply_defaulter_and_validator_interfaces.go

```diff
@@ -194,6 +189,8 @@ func validateSecretDestinations(k *functions.ResourceFunction, codeGenerationCon
 			"ValidateSecretDestinations"),
 	}
 
+	fn.AddReturn(astbuilder.QualifiedTypeName(astmodel.ControllerRuntimeAdmission.PackageName(), "Warnings"))
```

Comment:
I think you need to do a package lookup in `codeGenerationContext` here instead?

`fmtPackage := builder.codeGenerationContext.MustGetImportedPackageName(astmodel.FmtReference)`

What you have works because we happen to be importing the package w/ its package name, but in reality there's a risk of collisions and we might have something like:
`something github.com/controller-tools/controller-runtime/admissions`, where you need to use `something` and not `admissions` since it's been imported under a different name.

That's what the `MustGetImportedPackageName` helps with.

Reply:
This applies down below too.